### ISSUE TITLE
Mix team for tournaments

### ISSUE THREAD

**Issue by Paul255** [ScalaWilliam/ActionFPS#283]
We need to be able to accept mixed teams for tournaments, this would probably double the number of participant.

For example : w00p are not available to play this week but I want to play, Subjective is available so we create a team. But since we are not an official clan we can't play.

How can we do this ?

**Comment by ScalaWilliam** [comment]
Pls include source inspiration for issue (http://forum.cubers.net/thread-8814-post-172414.html#pid172414) in issue next time ;)

**Comment by ScalaWilliam** [comment]
Assumptions:
1. We want to benefit registered players as much as possible. Eg your games go towards your profile.
2. We want to minimise anonymous and random playing. Eg we can identify who is playing and avoid unknown super skilled people under aliases.
3. We want to focus on longer term vision. Which is the user authentication system in-game that enables a lot of power.
4. Currently: `game.team.player.user = users[user.current-nickname == game.team.player.nickname]` and `game.team.player.clan = clans[game.team.player.nickname matches clan.tag]`.
5. 1 & 2 => we want to retain a single user, regardless of which team they play in.

There are 2 solutions:
1. Short-term: adding multiple `nickname`s to `user`s which would allow multiple `clan`s.
2. Long-term: use authentication which presents us with `user` and `group` automatically out of the permitted list of `group`s.

## Solution 1: multiple `nickname`s per user
Management nightmare (multiple nicknames!), will need new work that does not fall in with long term vision.
We made the agreement previously that there'll only be one nickname at a time for a `user`.

## Solution 2: authentication
This is long term vision for ActionFPS. We need a client that supports authentication (near done) and server-side that supports authentication (near done). Here, a `user` and `group` is provided by the server automatically, and the user decides in the client side which `group` they want to play for at a game.

* User database generation and broadcast: https://github.com/ActionFPS/ActionFPS-Game/issues/33
* [AUTH] Include user metadata in the database update: https://github.com/ScalaWilliam/ActionFPS/issues/276
* Authentication Project: https://github.com/ActionFPS/ActionFPS-Game/projects/4
* https://github.com/ActionFPS/Incubator/blob/master/test_comb.py

There's long term beneficial work that we are **seeking help** for making happen, such as:
* Make automated build on Windows (AppVeyor): https://github.com/ActionFPS/ActionFPS-Game/issues/43
* OS X build instructions: https://github.com/ActionFPS/ActionFPS-Game/issues/45


## My conclusion

If players want mixed teams, they will have to help. Not interested in doing all the work for no gain.

**Comment by ScalaWilliam** [comment]
Satisfied with the answer @Paul255 ?

TLDR let's plug all efforts into the release here: https://github.com/ActionFPS/ActionFPS-Game/issues/48

**Comment by Paul255** [comment]
Yeah I understand, but is it possible to have a temporary solution, we could add 2-3 fake clan that people can use to register for tournaments ?

For example : 
 We add Mix.1 Mix.2 so if people want to play the tournament they can use this tag ?

**Comment by Sanzouille** [comment]
i disagree with the idea, we have the concept of clans and inters for a reason.

**Comment by ScalaWilliam** [comment]
Elaborate Sanzo.

**Comment by Sanzouille** [comment]
we built the architecture to be:
- inters are mixs with "lower" competition in form of individual player ranks. 
- clan wars -> tournaments higher competition, between fixed groups of people : clans.

**Comment by ScalaWilliam** [comment]
It appears we have a 'UK' team participating in the tournament as we already had it in the clans.

So either a Geographical or Clan grouping is acceptable but not anything else?

**Comment by Sanzouille** [comment]
I'd even deny the national teams. 
These teams on ly have an impact if they are used for special events. 
Right now it's just a poor way of bypassing the no mix rule. 
Problem is we have nothing to separate national from clans

**Comment by ScalaWilliam** [comment]
In AF authentication we made provision for 'groups', so perhaps for tournaments we'd stick only to groups that are clans?

I am against the idea of mixed teams too, don't see much cohesion going on there.

**Comment by Paul255** [comment]
Okay, I guess we can close this.

**Comment by ScalaWilliam** [comment]
Thanks :) 👍
